File: Java/src/main/java/Planes/experimentalPlane.java
package Planes;

import models.ClassificationLevel;
import models.ExperimentalTypes;

public class experimentalPlane extends Plane
{

    private ExperimentalTypes type;
    private ClassificationLevel classificationLevel;

    public ExperimentalPlane(String model, int maxSpeed, int maxFlightDistance, int maxLoadCapacity, ExperimentalTypes type, ClassificationLevel classificationLevel) 
    {
        super(model, maxSpeed, maxFlightDistance, maxLoadCapacity);
        this.type = type;
        this.classificationLevel = classificationLevel;
    }

    public ClassificationLevel GetClassificationLevel()
    {
        return classificationLevel;
    }

    public void SetClassificationLevel(ClassificationLevel classificationLevel)
    {
        this.classificationLevel = classificationLevel;
    }

    @Override
    public boolean Equals(Object o) 
    {
        return super.equals(o);
    }

    @Override
    public int HashCode()
    {
        return super.hashCode();
    }

    @Override
    public String ToString() 
    {
        return "experimentalPlane{" +
                "model='" + model + '\'' +
                '}';
    }
}

File: JS/Planes/experimentalPlane.js
const Plane = require('./Plane');

class experimentalPlane extends Plane
{

    constructor(model, maxSpeed, maxFlightDistance, maxLoadCapacity, type, classificationLevel)  
	{
        super(model, maxSpeed, maxFlightDistance, maxLoadCapacity);
        this.model = model;
        this.maxSpeed = maxSpeed;
        this.maxFlightDistance = maxFlightDistance;
        this.maxLoadCapacity = maxLoadCapacity;
        this.type = type;
        this.classificationLevel = classificationLevel;
    }


    GetModel()
    {
        return this.model;
    }

    SetModel(value) 
	{
        this.model = value;
    }

    GetMaxSpeed()
	{
        return this.maxSpeed;
    }

    SetMaxSpeed(value) 
	{
        this.MaxSpeed = value;
    }

    GetMaxFlightDistance() 
	{
        return this.maxFlightDistance;
    }

    SetMaxFlightDistance(value) 
	{
        this.maxFlightDistance = value;
    }

    GetMaxLoadCapacity() 
	{
        return this.maxLoadCapacity;
    }

    set maxLoadCapacity(value) 
	{
        this.maxLoadCapacity = value;
    }

    GetType() 
	{
        return this.Type;
    }

    SetType(value) 
	{
        this.type = value;
    }

    GetClassificationLevel() 
	{
        return this.classificationLevel;
    }

    SetClassificationLevel(value) 
	{
        this.classificationLevel = value;
    }
}

module.exports = experimentalPlane